src/Generator/SchemaGenerator.php

```diff
@@ -193,7 +194,17 @@ private function collectSerializationFields(Field $root, Variable $arrayVariable
             $fieldName       = $this->builder->val($propertyField->getName());
             $assignStatement = $this->builder->appendToAssociativeArray($arrayVariable, $fieldName, $value);
 
-            if ($propertyField->isOptional() && !$propertyField->isNullable()) {
+            if (
+                $propertyField->isOptional()
+                && $propertyField->isNullable()
+            ) {
+                throw new UnexpectedValueException('Optional nullable fields are not supported!');
+            }
+
+            if (
+                $propertyField->isOptional()
+                && !$propertyField->isNullable()
```

Comment:
This line can be deleted. It's always the case because of the previous condition.

Reply:
good point, fixed!